packages/neon-core/src/u/BigInteger.ts

```diff
@@ -0,0 +1,237 @@
+import { reverseHex } from "./misc";
+import BN from "bn.js";
+import { HexString } from "./HexString";
+
+/**
+ * This is a BigInteger representation, providing several helper methods to convert between different formats as well as basic math operations.
+ */
+export class BigInteger {
+  /** Internal holding value. */
+  #value: BN;
+
+  /**
+   * Creates a BigInteger from a twos complement hexstring.
+   * @param hexstring - HexString class or a bigendian string.
+   *
+   * @example
+   * const negativeOne = BigInteger.fromTwos("ff");
+   * const oneTwoEight = BigInteger.fromTwos("0080");
+   *
+   * // Apply a second parameter when using little endian.
+   * const oneTwoEightAgain = BigInteger.fromTwos("8000", true);
+   */
+  public static fromTwos(hexstring: HexString): BigInteger;
+  public static fromTwos(hexstring: string, littleEndian?: boolean): BigInteger;
+  public static fromTwos(
+    hexstring: string | HexString,
+    littleEndian = false
+  ): BigInteger {
+    const cleanedInput =
+      hexstring instanceof HexString
+        ? hexstring
+        : HexString.fromHex(hexstring, littleEndian);
+    return new BigInteger(
+      new BN(cleanedInput.toBigEndian(), 16).fromTwos(
+        cleanedInput.byteLength * 8
+      )
+    );
+  }
+
+  /**
+   * Creates a BigInteger from a hexstring.
+   * @param hexstring - HexString class or a bigendian string.
```

Comment:
```suggestion
   * @param hexstring - HexString class or a little endian string.
```
same as `fromTwos()`